vars/buildPlugin.groovy

```diff
@@ -142,29 +126,23 @@ def call(Map params = [:]) {
                         }
 
                         stage("Archive (${stageIdentifier})") {
-                            if (isMaven && archiveFindbugs) {
-                                def fp = [pattern: params?.findbugs?.pattern ?: '**/target/findbugsXml.xml']
-                                if (params?.findbugs?.unstableNewAll) {
-                                    fp['unstableNewAll'] = "${params.findbugs.unstableNewAll}"
-                                }
-                                if (params?.findbugs?.unstableTotalAll) {
-                                    fp['unstableTotalAll'] = "${params.findbugs.unstableTotalAll}"
-                                }
-                                findbugs(fp)
-                            }
-                            if (isMaven && archiveCheckstyle) {
-                                def cp = [pattern: params?.checkstyle?.pattern ?: '**/target/checkstyle-result.xml']
-                                if (params?.checkstyle?.unstableNewAll) {
-                                    cp['unstableNewAll'] = "${params.checkstyle.unstableNewAll}"
-                                }
-                                if (params?.checkstyle?.unstableTotalAll) {
-                                    cp['unstableTotalAll'] = "${params.checkstyle.unstableTotalAll}"
-                                }
-                                checkstyle(cp)
-                            }
                             if (failFast && currentBuild.result == 'UNSTABLE') {
                                 error 'There were test failures; halting early'
                             }
+
+                            if (first) {
```

Comment:
IIUC, the current condition will be always `true`, so I wonder if this condition is redundant somehow

Reply:
Isn't the condition only true for the first configuration we build? E.g. for linux it is true but for windowsit isfalse`?

Reply:
Gotcha!! Thanks for the clarification, I actually have to look it at twice, thanks again :)